### PR TITLE
MOTECH-2717 Using XLMNS + APP_ID as an unique identifier, and changed tests accordingly

### PR DESCRIPTION
**Pull request by mjazdzdewski**
…

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: commcare/src/test/java/org/motechproject/commcare/util/ResponseXML.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public static String getFormXML() {
                 "      version=\"41\"\n" +
                 "      name=\"" + FORM_NAME + "\"\n" +
                 "      xmlns:jrm=\"http://dev.commcarehq.org/jr/xforms\"\n" +
-                "      xmlns=\"" + DummyCommcareSchema.XMLNS1 + "\">\n" +
+                "      xmlns=\"" + DummyCommcareSchema.XMLNS1 + DummyCommcareSchema.APP_ID1 + "\">\n" +
```
Comment (kmadej) [inline commcare/src/test/java/org/motechproject/commcare/util/ResponseXML.java]: app id should not be included in xmlns value

**Review by kmadej — CHANGES_REQUESTED** [review]

File: commcare/src/main/java/org/motechproject/commcare/tasks/builder/FormTriggerBuilder.java [hunk 2]
```diff
@@ -99,9 +99,13 @@ private TriggerEventRequest buildTriggerForForm(Config config, CommcareApplicati
...
 
         String displayName = DisplayNameHelper.buildDisplayName(RECEIVED_FORM, formName, applicationName, config.getName());
-
-        return new TriggerEventRequest(displayName, FORMS_EVENT + "." + config.getName() + "." + form.getXmlns(),
-                null, buildTriggerParameters(form), FORMS_EVENT);
+        if (application.getCommcareAppId() != null) {
+            return new TriggerEventRequest(displayName, FORMS_EVENT + "." + config.getName() + "." + form.getXmlns() + application.getCommcareAppId(),
+                    null, buildTriggerParameters(form), FORMS_EVENT);
+        } else {
+            return new TriggerEventRequest(displayName, FORMS_EVENT + "." + config.getName() + "." + form.getXmlns(),
```
Comment (kmadej) [inline commcare/src/main/java/org/motechproject/commcare/tasks/builder/FormTriggerBuilder.java]: first of all,  there should be one return statement. Secondly it should be better to verify if commcareAppId is null in one line

File: commcare/src/main/java/org/motechproject/commcare/tasks/builder/FormActionBuilder.java [hunk 3]
```diff
@@ -61,9 +61,13 @@ public FormActionBuilder(CommcareSchemaService schemaService, CommcareConfigServ
 
                     String displayName = DisplayNameHelper.buildDisplayName(DisplayNames.SUBMIT_FORM, form.getFormName(),
                             application.getApplicationName(), config.getName());
+                    String appId = application.getCommcareAppId();
```
Comment (kmadej) [inline commcare/src/main/java/org/motechproject/commcare/tasks/builder/FormActionBuilder.java]: this should be done in one line

File: commcare/src/main/java/org/motechproject/commcare/tasks/builder/FormActionBuilder.java [hunk 4]
```diff
@@ -100,6 +105,14 @@ public FormActionBuilder(CommcareSchemaService schemaService, CommcareConfigServ
...
         }
 
+        builder = new ActionParameterRequestBuilder();
+        builder.setValue(form.getXmlns())
+                .setDisplayName("xmlns")
+                .setKey("xmlns")
+                .setOrder(order++)
+                .setType(ParameterType.UNICODE.getValue())
+                .setHidden(true);
+        parameters.add(builder.createActionParameterRequest());
```
Comment (kmadej) [inline commcare/src/main/java/org/motechproject/commcare/tasks/builder/FormActionBuilder.java]: move this logic into private method 'addXmlnsParameter'

File: commcare/src/main/java/org/motechproject/commcare/service/impl/CommcareFormsEventParser.java [hunk 5]
```diff
@@ -54,7 +54,12 @@ public String parseEventSubject(String eventSubject, Map<String, Object> eventPa
 
         if (eventSubject.equals(FORMS_EVENT)) {
             String xmlns = (String) ((Map) eventParameters.get(ATTRIBUTES)).get("xmlns");
-            return eventSubject.concat(".").concat(configName).concat(".").concat(xmlns);
+            String appId = (String) ((Map) eventParameters.get(ATTRIBUTES)).get("app_id");
+            if (appId != null) {
+                return eventSubject.concat(".").concat(configName).concat(".").concat(xmlns).concat(appId);
+            } else {
+                return eventSubject.concat(".").concat(configName).concat(".").concat(xmlns);
+            }
```
Comment (kmadej) [inline commcare/src/main/java/org/motechproject/commcare/service/impl/CommcareFormsEventParser.java:65]: there should be one return statement, please also simplify this code. Additionally add warning that app_id is null

**Review by kmadej — CHANGES_REQUESTED** [review]

**Review by kmadej — APPROVED** [review]

**Pull request merged**